pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -569,6 +569,66 @@ func (c *Cluster) handleRemovedNodes(config *provisionConfig) {
 	logger.Infof("done processing removed nodes")
 }
 
+func (c *Cluster) cleanupTimeoutNode(selector string, config *provisionConfig, nodeName, crushName string) {
+	// update the orchestration status of this removed node to the completed state
+	if err := c.updateOSDStatus(nodeName, OrchestrationStatus{Status: OrchestrationStatusCompleted}); err != nil {
+		config.addError("failed to set orchestration completed status for timeout node %s: %+v", nodeName, err)
+		return
+	}
+
+	// remove all batch jobs on this node
+	osdProps := osdProperties{
+		crushHostname: nodeName,
+		devices:       []rookalpha.Device{},
+		selection:     rookalpha.Selection{DeviceFilter: "none"},
+		resources:     v1.ResourceRequirements{},
+		storeConfig:   osdconfig.StoreConfig{},
+	}
+
+	job, err := c.makeJob(osdProps)
+	if err != nil {
+		message := fmt.Sprintf("failed to create prepare job node %s: %+v", nodeName, err)
+		config.addError(message)
+		return
+	}
+
+	if !c.runJob(job, nodeName, config, "remove") {
```

Comment:
I don't think we can assume OSDs should always be removed when they timeout. This might be in a subsequent orchestration where the osd is already configured and the orchestration just timed out on this node. We wouldn't want to clean up the OSDs in that case.

Reply:
We need to know if the node really be removed from the cluster/K8s and then we go to remove the OSDs.